archiver.go

```diff
@@ -390,7 +394,7 @@ func ByExtension(filename string) (interface{}, error) {
 	case *Xz:
 		return NewXz(), nil
 	}
-	return nil, fmt.Errorf("format unrecognized by filename: %s", filename)
+	return nil, ErrFormatNotRecognized
```

Comment:
This is a different error, even though the wording is similar, and shouldn't be changed.